# Team VizLaLune: Milano-Cortina 2026 Olympic Winter Games


| Student's name | SCIPER |
| -------------- | ------ |
| Thomas Lenges  | 325245 |
| Léontine Lefranc | 345240 |
| Manon Darnaud | 361028 |

This Jupyter Notebook serves to assess the quality of the data from the Kaggle dataset *["Milano-Cortina 2026 Olympic Winter Games"](https://www.kaggle.com/datasets/piterfm/milano-cortina-2026-olympic-winter-games/data?select=medals.csv)*.

This data exploration aims to evaluate how much preprocessing and data cleaning are required before proceeding with visualization.

The exploration leads to a preprocessing step that produces final `.csv` files to be used for visualization (see *[PREPROCESSING_Milano_Cortina_2026_Olympic_Winter_Games.ipynb](https://github.com/com-480-data-visualization/VizLaLune)*)

The dataset has the following files:

- athletes.csv
- coaches.csv
- medallists.csv
- medals.csv
- schedules.csv
- teams.csv
- venues.csv

# Download dataset

In [78]:
# Download dataset from Kaggle using kagglehub library
import kagglehub

path = kagglehub.dataset_download("piterfm/milano-cortina-2026-olympic-winter-games")

print("Path to dataset files:", path)

Path to dataset files: /home/thomas/.cache/kagglehub/datasets/piterfm/milano-cortina-2026-olympic-winter-games/versions/11


# Initial dataset investigation

In [79]:
# To avoid re-download the dataset (need to be changed by user)
#path = "/home/thomas/.cache/kagglehub/datasets/piterfm/milano-cortina-2026-olympic-winter-games/versions/11"

In [80]:
# Load the dataset using pandas
import pandas as pd

df_athletes = pd.read_csv(path + "/athletes.csv")
df_coaches = pd.read_csv(path + "/coaches.csv")
df_medallists = pd.read_csv(path + "/medallists.csv")
df_medals = pd.read_csv(path + "/medals.csv")
df_schedules = pd.read_csv(path + "/schedules.csv")
df_teams = pd.read_csv(path + "/teams.csv")
df_venues = pd.read_csv(path + "/venues.csv")

From official info: the 2026 Winter Olympics featured 16 disciplines and 116 events for a total of 348 medals (key values one can verify in different sub-datasets)

## Athletes

In [81]:
df_athletes.shape

(2916, 6)

In [82]:
duplicated_athletes = df_athletes[df_athletes.duplicated()]
duplicated_athletes

,code,name,gender,country_code,flag_bearer,events


In [83]:
df_athletes.nunique()

code            2916
name            2916
gender             2
country_code      93
flag_bearer        4
events           344
dtype: int64

In [84]:
df_athletes.info()

<class 'pandas.DataFrame'>
RangeIndex: 2916 entries, 0 to 2915
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   code          2916 non-null   int64
 1   name          2916 non-null   str  
 2   gender        2916 non-null   str  
 3   country_code  2916 non-null   str  
 4   flag_bearer   2916 non-null   int64
 5   events        2916 non-null   str  
dtypes: int64(2), str(4)
memory usage: 136.8 KB


In [85]:
df_athletes.head()

,code,name,gender,country_code,flag_bearer,events
0,139893,AABREKK Ingrid Bergene,F,NOR,0,"[{'discipline': 'Cross-Country Skiing', 'event..."
1,24464,AAGAARD Mikkel,M,DEN,0,"[{'discipline': 'Ice Hockey', 'event': 'Men'}]"
2,45083,AALTO Antti,M,FIN,0,"[{'discipline': 'Ski Jumping', 'event': ""Men's..."
3,32319,ABATANGELO Aurora,F,ITA,0,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
4,4972,ABDI Fayik,M,KSA,0,"[{'discipline': 'Alpine Skiing', 'event': ""Men..."


In [86]:
df_athletes.tail()

,code,name,gender,country_code,flag_bearer,events
2911,25116,ZUMWINKLE Grace,F,USA,0,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
2912,44328,ZUNTE Darta,F,EST,0,"[{'discipline': 'Skeleton', 'event': 'Women'}]"
2913,7319,ZURBAY Anabelle,F,IRL,1,"[{'discipline': 'Alpine Skiing', 'event': ""Wom..."
2914,19124,ZURBRIGGEN Anina,F,BUL,0,"[{'discipline': 'Alpine Skiing', 'event': ""Wom..."
2915,23401,ZUREK Damian,M,POL,0,"[{'discipline': 'Speed Skating', 'event': ""Men..."


In [87]:
df_athletes.describe()

,code,flag_bearer
count,2916.000000,2916.000000
mean,36401.582990,0.122085
std,16043.661964,0.470224
min,4098.000000,0.000000
25%,24173.000000,0.000000
50%,39906.500000,0.000000
75%,45897.000000,0.000000
max,170798.000000,3.000000


In [88]:
import ast

# Checking if 116 unique events in athletes.csv is correct (some have "" as their event name)
athlete_events = set()
empty_event_count = 0

for e in df_athletes["events"]:
    events_list = ast.literal_eval(e)   # convert string -> list of dicts
    for event in events_list:
        athlete_events.add((event["discipline"], event["event"]))

empty_events = [pair for pair in athlete_events if pair[1] == ""]

print("Unique events in athletes.csv:", len(athlete_events))
print("Empty events in athletes.csv:", len(empty_events))

Unique events in athletes.csv: 121
Empty events in athletes.csv: 5


Conclusion:

Complete dataset for athletes. 121-5 = 116 events (all events featured). Non duplicated dataset for athletes. Extra flag bearer info.

More events than the 116 official number because some athletes combine different events (Need for pre-processing)!

## Coaches

In [89]:
df_coaches.shape

(50, 7)

In [90]:
duplicated_coaches = df_coaches[df_coaches.duplicated()]
duplicated_coaches

,code,name,gender,country_code,nationality,function,events


In [91]:
df_coaches.nunique()

code            50
name            50
gender           2
country_code    18
nationality     15
function         2
events           8
dtype: int64

In [92]:
df_coaches.info()

<class 'pandas.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   code          50 non-null     int64
 1   name          50 non-null     str  
 2   gender        50 non-null     str  
 3   country_code  50 non-null     str  
 4   nationality   50 non-null     str  
 5   function      50 non-null     str  
 6   events        50 non-null     str  
dtypes: int64(1), str(6)
memory usage: 2.9 KB


In [93]:
df_coaches.head()

,code,name,gender,country_code,nationality,function,events
0,50140,BOUCHARD Eric,M,ITA,CAN,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
1,50988,CERNOVSKY Vladimir,M,CZE,CZE,Coach,"[{'discipline': 'Curling', 'event': 'Mixed Dou..."
2,49020,CHARETTE Pierre,M,SUI,CAN,Coach,"[{'discipline': 'Curling', 'event': 'Women'}]"
3,53809,COOPER Jon,M,CAN,CAN,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Men'}]"
4,49035,de CRUZ Peter,M,SUI,SUI,Coach,"[{'discipline': 'Curling', 'event': 'Mixed Dou..."


In [94]:
df_coaches.tail()

,code,name,gender,country_code,nationality,function,events
45,47166,VITOLINS Harijs,M,LAT,LAT,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Men'}]"
46,22959,WALSTAD Steffen,M,EST,NOR,Coach,"[{'discipline': 'Curling', 'event': 'Mixed Dou..."
47,53327,WEBSTER Paul,M,CAN,CAN,Coach,"[{'discipline': 'Curling', 'event': 'Men'}]"
48,25178,WROBLEWSKI John,M,USA,USA,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
49,39951,ZANG Jialiang,M,CHN,CHN,Coach,"[{'discipline': 'Curling', 'event': 'Women'}]"


In [95]:
disciplines = set()

for e in df_coaches["events"]:
    events_list = ast.literal_eval(e)   # convert string -> list of dicts
    for event in events_list:
        disciplines.add(event["discipline"])

disciplines

{'Curling', 'Ice Hockey'}

Conclusion:

Incomplete dataset for coaches (only 50). Only curling and ice hockey are represented. Limited potential use.

## Medallists

In [96]:
df_medallists.shape

(709, 11)

In [97]:
duplicated_medallists = df_medallists[df_medallists.duplicated()]
duplicated_medallists

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name


In [98]:
df_medallists.nunique()

date                16
medal_code           3
medal                3
code               534
name               534
gender               2
country_code        30
country             30
discipline          16
discipline_code     16
event_name         104
dtype: int64

In [99]:
df_medallists.info()

<class 'pandas.DataFrame'>
RangeIndex: 709 entries, 0 to 708
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   date             709 non-null    str  
 1   medal_code       709 non-null    int64
 2   medal            709 non-null    str  
 3   code             709 non-null    int64
 4   name             709 non-null    str  
 5   gender           709 non-null    str  
 6   country_code     709 non-null    str  
 7   country          709 non-null    str  
 8   discipline       709 non-null    str  
 9   discipline_code  709 non-null    str  
 10  event_name       709 non-null    str  
dtypes: int64(2), str(9)
memory usage: 61.1 KB


In [100]:
df_medallists.head()

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name
0,2026-02-07,1,GOLD,48729,von ALLMEN Franjo,M,SUI,Switzerland,Alpine Skiing,ALP,Men's Downhill
1,2026-02-07,2,SILVER,25998,FRANZONI Giovanni,M,ITA,Italy,Alpine Skiing,ALP,Men's Downhill
2,2026-02-07,3,BRONZE,26002,PARIS Dominik,M,ITA,Italy,Alpine Skiing,ALP,Men's Downhill
3,2026-02-07,1,GOLD,30739,KARLSSON Frida,F,SWE,Sweden,Cross-Country Skiing,CCS,Women's 10km + 10km Skiathlon
4,2026-02-07,2,SILVER,30707,ANDERSSON Ebba,F,SWE,Sweden,Cross-Country Skiing,CCS,Women's 10km + 10km Skiathlon


In [101]:
df_medallists.tail()

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name
704,2026-02-22,2,SILVER,53901,SUZUKI Nick,M,CAN,Canada,Ice Hockey,IHO,Men
705,2026-02-22,2,SILVER,53905,THEODORE Shea,M,CAN,Canada,Ice Hockey,IHO,Men
706,2026-02-22,2,SILVER,53907,THOMPSON Logan,M,CAN,Canada,Ice Hockey,IHO,Men
707,2026-02-22,2,SILVER,53910,TOEWS Devon,M,CAN,Canada,Ice Hockey,IHO,Men
708,2026-02-22,2,SILVER,53914,WILSON Tom,M,CAN,Canada,Ice Hockey,IHO,Men


In [102]:
# Checking if 116 unique events in medallists.csv is correct
print("Nb different events:" + str(df_medallists.groupby(["discipline", "event_name"]).ngroups))

Nb different events:116


In [103]:
df_medallists.groupby("name").size()

name
ADAMCZYKOVA Eva         1
AEBERHARD Mario         1
AHO Sebastian           1
AICHER Emma             2
ALONSO RODRIGUEZ Ana    2
                       ..
de LAAT Itzhak          1
den DULK Tineke         1
van 'T WOUT Jens        4
van 'T WOUT Melle       2
von ALLMEN Franjo       3
Length: 534, dtype: int64

In [104]:
# Check if each event has all medal types
medals_per_event = df_medallists.groupby(["discipline", "event_name"])["medal"].nunique().reset_index()

print("Distribution of medal types per event:")
print(medals_per_event["medal"].value_counts())

Distribution of medal types per event:
medal
3    114
2      2
Name: count, dtype: int64


Due to ties (see below for proof)

In [105]:
# Show events with only 2 medals
two_medal_events = medals_per_event[medals_per_event["medal"] == 2]
two_medal_events


,discipline,event_name,medal
4,Alpine Skiing,Men's Team Combined,2
6,Alpine Skiing,Women's Giant Slalom,2


In [106]:
df_medallists[df_medallists["event_name"] == "Men's Team Combined"]

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name
66,2026-02-09,1,GOLD,48729,von ALLMEN Franjo,M,SUI,Switzerland,Alpine Skiing,ALP,Men's Team Combined
67,2026-02-09,1,GOLD,48602,NEF Tanguy,M,SUI,Switzerland,Alpine Skiing,ALP,Men's Team Combined
68,2026-02-09,2,SILVER,48571,MEILLARD Loic,M,SUI,Switzerland,Alpine Skiing,ALP,Men's Team Combined
69,2026-02-09,2,SILVER,48607,ODERMATT Marco,M,SUI,Switzerland,Alpine Skiing,ALP,Men's Team Combined
70,2026-02-09,2,SILVER,20284,FELLER Manuel,M,AUT,Austria,Alpine Skiing,ALP,Men's Team Combined
71,2026-02-09,2,SILVER,20306,KRIECHMAYR Vincent,M,AUT,Austria,Alpine Skiing,ALP,Men's Team Combined


In [107]:
df_medallists[df_medallists["event_name"] == "Women's Giant Slalom"]

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name
269,2026-02-15,1,GOLD,6464,BRIGNONE Federica,F,ITA,Italy,Alpine Skiing,ALP,Women's Giant Slalom
270,2026-02-15,2,SILVER,42237,STJERNESUND Thea Louise,F,NOR,Norway,Alpine Skiing,ALP,Women's Giant Slalom
271,2026-02-15,2,SILVER,27515,HECTOR Sara,F,SWE,Sweden,Alpine Skiing,ALP,Women's Giant Slalom


Conclusion: 

Complete sub-dataset!

More medallists than events due to team events.

Medallists are named once per medal.

## Medals

In [108]:
duplicated_medals = df_medals[df_medals.duplicated()]
duplicated_medals

,country,country_code,gold,silver,bronze,total,rank,rank_total


In [109]:
df_medals.nunique()

country         29
country_code    29
gold            10
silver          11
bronze          12
total           17
rank            26
rank_total      17
dtype: int64

In [110]:
df_medals.shape

(29, 8)

In [111]:
df_medals.info()

<class 'pandas.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   country       29 non-null     str  
 1   country_code  29 non-null     str  
 2   gold          29 non-null     int64
 3   silver        29 non-null     int64
 4   bronze        29 non-null     int64
 5   total         29 non-null     int64
 6   rank          29 non-null     int64
 7   rank_total    29 non-null     int64
dtypes: int64(6), str(2)
memory usage: 1.9 KB


In [112]:
df_medals["total"].sum()

np.int64(348)

In [113]:
df_medals.head()

,country,country_code,gold,silver,bronze,total,rank,rank_total
0,Norway,NOR,18,12,11,41,1,1
1,United States,USA,12,12,9,33,2,2
2,Netherlands,NED,10,7,3,20,3,9
3,Italy,ITA,10,6,14,30,4,3
4,Germany,GER,8,10,8,26,5,4


In [114]:
df_medals.tail()

,country,country_code,gold,silver,bronze,total,rank,rank_total
24,Denmark,DEN,0,1,0,1,25,24
25,Estonia,EST,0,1,0,1,25,24
26,Georgia,GEO,0,1,0,1,25,24
27,Bulgaria,BUL,0,0,2,2,28,22
28,Belgium,BEL,0,0,1,1,29,24


In [115]:
df_medals.describe()

,gold,silver,bronze,total,rank,rank_total
count,29.000000,29.000000,29.000000,29.000000,29.000000,29.000000
mean,4.000000,4.034483,3.965517,12.000000,14.862069,14.241379
std,4.511889,3.877432,4.083790,11.655777,8.356880,7.744377
min,0.000000,0.000000,0.000000,1.000000,1.000000,1.000000
25%,0.000000,1.000000,1.000000,2.000000,8.000000,8.000000
50%,3.000000,2.000000,2.000000,6.000000,15.000000,14.000000
75%,6.000000,7.000000,6.000000,21.000000,22.000000,22.000000
max,18.000000,12.000000,14.000000,41.000000,29.000000,24.000000


Conclusion:

Similar ranks due to ties. Can also exploit rank_total(medal sum instead of first gold medal sum then silver then bronze) instead of rank. 

Only one medal per sport not per individual (709 vs. 348)

## Schedules

In [116]:
duplicated_schedule = df_schedules[df_schedules.duplicated()]
duplicated_schedule

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
7,2026-02-04T11:30:00+01:00,2026-02-04T13:30:00+01:00,2026-02-04,FINISHED,Alpine Skiing,ALP,Men's Downhill,0,Men's Downhill Official Training,M,INDV,Stelvio Ski Centre,SSC,Stelvio Ski Centre-Alpine Skiing Course,SAL,ALPMDH----------------TRNO000100--
8,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet A,CUA,CURXTEAM2-------------PREL000101--
9,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet B,CUB,CURXTEAM2-------------PREL000102--
10,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet C,CUC,CURXTEAM2-------------PREL000103--
11,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet D,CUD,CURXTEAM2-------------PREL000104--
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1879,2026-02-22T11:09:00+01:00,2026-02-22T11:34:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,0,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000200--
1880,2026-02-22T11:38:00+01:00,2026-02-22T12:03:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,1,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000300--
1881,2026-02-22T12:15:00+01:00,2026-02-22T13:00:00+01:00,2026-02-22,FINISHED,Bobsleigh,BOB,4-man,1,4-man Competition,M,TEAM,Cortina Sliding Centre,CSC,Cortina Sliding Centre,CSC,BOBMTEAM4-------------FNL-000400--
1882,2026-02-22T14:10:00+01:00,2026-02-22T16:40:00+01:00,2026-02-22,FINISHED,Ice Hockey,IHO,Men,1,Men's Play-off Round,M,TEAM,Milano Santagiulia Ice Hockey Arena,MH1,Milano Santagiulia Ice Hockey Arena,MH1,IHOMTEAM6-------------FNL-000100--


In [117]:
df_schedules = df_schedules.drop_duplicates()
df_schedules.shape

(628, 16)

In [118]:
df_schedules.nunique()

start_date         485
end_date           503
day                 19
status               3
discipline          17
discipline_code     17
event              112
event_medal          3
phase              245
gender               4
event_type           5
venue               15
venue_code          15
location            25
location_code       25
id                 628
dtype: int64

In [119]:
df_schedules.info()

<class 'pandas.DataFrame'>
Index: 628 entries, 0 to 1843
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   start_date       628 non-null    str  
 1   end_date         626 non-null    str  
 2   day              628 non-null    str  
 3   status           628 non-null    str  
 4   discipline       628 non-null    str  
 5   discipline_code  628 non-null    str  
 6   event            628 non-null    str  
 7   event_medal      628 non-null    int64
 8   phase            628 non-null    str  
 9   gender           628 non-null    str  
 10  event_type       628 non-null    str  
 11  venue            628 non-null    str  
 12  venue_code       628 non-null    str  
 13  location         628 non-null    str  
 14  location_code    628 non-null    str  
 15  id               628 non-null    str  
dtypes: int64(1), str(15)
memory usage: 83.4 KB


In [120]:
df_schedules.head()

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
0,2026-02-04T11:30:00+01:00,2026-02-04T13:30:00+01:00,2026-02-04,FINISHED,Alpine Skiing,ALP,Men's Downhill,0,Men's Downhill Official Training,M,INDV,Stelvio Ski Centre,SSC,Stelvio Ski Centre-Alpine Skiing Course,SAL,ALPMDH----------------TRNO000100--
1,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet A,CUA,CURXTEAM2-------------PREL000101--
2,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet B,CUB,CURXTEAM2-------------PREL000102--
3,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet C,CUC,CURXTEAM2-------------PREL000103--
4,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet D,CUD,CURXTEAM2-------------PREL000104--


In [121]:
df_schedules.tail()

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
1839,2026-02-22T11:09:00+01:00,2026-02-22T11:34:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,0,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000200--
1840,2026-02-22T11:38:00+01:00,2026-02-22T12:03:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,1,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000300--
1841,2026-02-22T12:15:00+01:00,2026-02-22T13:00:00+01:00,2026-02-22,FINISHED,Bobsleigh,BOB,4-man,1,4-man Competition,M,TEAM,Cortina Sliding Centre,CSC,Cortina Sliding Centre,CSC,BOBMTEAM4-------------FNL-000400--
1842,2026-02-22T14:10:00+01:00,2026-02-22T16:40:00+01:00,2026-02-22,FINISHED,Ice Hockey,IHO,Men,1,Men's Play-off Round,M,TEAM,Milano Santagiulia Ice Hockey Arena,MH1,Milano Santagiulia Ice Hockey Arena,MH1,IHOMTEAM6-------------FNL-000100--
1843,2026-02-22T20:30:00+01:00,2026-02-22T23:00:00+01:00,2026-02-22,SCHEDULED,Ceremonies,CER,Closing Ceremony,0,Closing Ceremony,G,NONE,Verona Olympic Arena,VOA,Verona Olympic Arena,VOA,CERGCLOSING-----------OTHR--------


In [122]:
df_schedules.describe()

,event_medal
count,628.000000
mean,0.237261
std,0.535276
min,0.000000
25%,0.000000
50%,0.000000
75%,0.000000
max,3.000000


In [123]:
df_schedules.groupby("status").size()

status
CANCELLED      3
FINISHED     624
SCHEDULED      1
dtype: int64

In [124]:
phase_cancelled = df_schedules[df_schedules["status"] == "CANCELLED"]["phase"]
phase_cancelled

18      Women's Downhill Official Training
1246                      Men's Super Team
1353           Women's Team Pursuit Finals
Name: phase, dtype: str

In [125]:
df_schedules[df_schedules["phase"].isin(phase_cancelled)]

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
18,2026-02-05T11:30:00+01:00,NaN,2026-02-05,CANCELLED,Alpine Skiing,ALP,Women's Downhill,0,Women's Downhill Official Training,W,INDV,Tofane Alpine Skiing Centre,CAL,Tofane Alpine Skiing Centre,CAL,ALPWDH----------------TRNO000100--
76,2026-02-06T11:30:00+01:00,2026-02-06T13:30:00+01:00,2026-02-06,FINISHED,Alpine Skiing,ALP,Women's Downhill,0,Women's Downhill Official Training,W,INDV,Tofane Alpine Skiing Centre,CAL,Tofane Alpine Skiing Centre,CAL,ALPWDH----------------TRNO000200--
132,2026-02-07T11:30:00+01:00,2026-02-07T13:30:00+01:00,2026-02-07,FINISHED,Alpine Skiing,ALP,Women's Downhill,0,Women's Downhill Official Training,W,INDV,Tofane Alpine Skiing Centre,CAL,Tofane Alpine Skiing Centre,CAL,ALPWDH----------------TRNO000300--
1237,2026-02-16T18:00:00+01:00,2026-02-16T18:35:00+01:00,2026-02-16,FINISHED,Ski Jumping,SJP,Men's Super Team,0,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-TR01SJ--
1239,2026-02-16T19:00:00+01:00,2026-02-16T19:35:00+01:00,2026-02-16,FINISHED,Ski Jumping,SJP,Men's Super Team,0,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-0001SJ--
1244,2026-02-16T19:50:00+01:00,2026-02-16T20:16:00+01:00,2026-02-16,FINISHED,Ski Jumping,SJP,Men's Super Team,1,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-0002SJ--
1246,2026-02-16T20:31:00+01:00,NaN,2026-02-16,CANCELLED,Ski Jumping,SJP,Men's Super Team,0,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-0003SJ--
1353,2026-02-17T15:43:00+01:00,2026-02-17T15:49:00+01:00,2026-02-17,CANCELLED,Speed Skating,SSK,Women's Team Pursuit,0,Women's Team Pursuit Finals,W,TEAM,Milano Speed Skating Stadium,MSS,Milano Speed Skating Stadium,MSS,SSKWTEAMPU------------FNL-D00100--
1354,2026-02-17T15:49:00+01:00,2026-02-17T15:55:00+01:00,2026-02-17,FINISHED,Speed Skating,SSK,Women's Team Pursuit,0,Women's Team Pursuit Finals,W,TEAM,Milano Speed Skating Stadium,MSS,Milano Speed Skating Stadium,MSS,SSKWTEAMPU------------FNL-C00100--
1358,2026-02-17T16:41:00+01:00,2026-02-17T16:47:00+01:00,2026-02-17,FINISHED,Speed Skating,SSK,Women's Team Pursuit,3,Women's Team Pursuit Finals,W,TEAM,Milano Speed Skating Stadium,MSS,Milano Speed Skating Stadium,MSS,SSKWTEAMPU------------FNL-B00100--


In [126]:
df_schedules = df_schedules[~df_schedules["status"].isin(["CANCELLED", "SCHEDULED"])]
df_schedules.shape

(624, 16)

In [127]:
# Check if all 116 events are within schedule.csv
df_schedules.groupby(["discipline", "event"]).ngroups

123

In [128]:
df_schedules[df_schedules["event"].str.contains("ceremon", case=False, na=False)]

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
86,2026-02-06T20:00:00+01:00,2026-02-06T22:30:00+01:00,2026-02-06,FINISHED,Ceremonies,CER,Opening Ceremony,0,Opening Ceremony,G,NONE,Multiple Venues,MTP,Multiple Venues,MTP,CERGOPENING-----------OTHR--------


In [129]:
# Actually there are other ceremony events!
df_schedules = df_schedules[~df_schedules["event"].str.contains("ceremon", case=False)]
df_schedules.groupby(["discipline", "event"]).ngroups

122

In [130]:
pd.set_option('display.max_rows', None)
df_schedules.groupby(["discipline", "event"]).size().reset_index(name="count")

,discipline,event,count
0,Alpine Skiing,Men's Downhill,4
1,Alpine Skiing,Men's Giant Slalom,2
2,Alpine Skiing,Men's Slalom,2
3,Alpine Skiing,Men's Super-G,1
4,Alpine Skiing,Men's Team Combined,2
5,Alpine Skiing,Women's Downhill,3
6,Alpine Skiing,Women's Giant Slalom,2
7,Alpine Skiing,Women's Slalom,2
8,Alpine Skiing,Women's Super-G,1
9,Alpine Skiing,Women's Team Combined,3


In [131]:
pd.reset_option('display.max_rows')

Used this large table to understand where the 6 too many events come from

Some events are named with training:

-Nordic Combined	Large Hill Training instead of Nordic Combined	Individual Gundersen Large Hill/10km

-Nordic Combined	Normal Hill Training instedad of Nordic Combined	Individual Gundersen Normal Hill/10km

Also some events just have different nomenclature names:

-Men's LH Individual vs. Men's Large Hill

-Men's NH Individual vs. Men's Normal Hill

-Women's LH Individual vs. Women's Large Hill

-Women's NH Individual vs. Women's Normal Hill

In [132]:
df_schedules["event"] = df_schedules["event"].replace("Large Hill Training", "Individual Gundersen Large Hill/10km")
df_schedules["event"] = df_schedules["event"].replace("Normal Hill Training", "Individual Gundersen Normal Hill/10km")
df_schedules["event"] = df_schedules["event"].replace("Men's Large Hill", "Men's LH Individual")
df_schedules["event"] = df_schedules["event"].replace("Men's Normal Hill", "Men's NH Individual")
df_schedules["event"] = df_schedules["event"].replace("Women's Large Hill", "Women's LH Individual")
df_schedules["event"] = df_schedules["event"].replace("Women's Normal Hill", "Women's NH Individual")
df_schedules.groupby(["discipline", "event"]).ngroups

116

In [133]:
# Count unique events (event_medal == 1)
medal_counts = df_schedules.groupby(["discipline", "event", "event_medal"]).size().reset_index(name="nb_rows")
medal_counts[medal_counts["event_medal"] == 1]

,discipline,event,event_medal,nb_rows
1,Alpine Skiing,Men's Downhill,1,1
3,Alpine Skiing,Men's Giant Slalom,1,1
5,Alpine Skiing,Men's Slalom,1,1
6,Alpine Skiing,Men's Super-G,1,1
8,Alpine Skiing,Men's Team Combined,1,1
...,...,...,...,...
201,Speed Skating,Women's 3000m,1,1
202,Speed Skating,Women's 5000m,1,1
203,Speed Skating,Women's 500m,1,1
205,Speed Skating,Women's Mass Start,1,1


In [134]:
# Count unique events (event_medal > 1)
medal_counts = df_schedules.groupby(["discipline", "event", "event_medal"]).size().reset_index(name="nb_rows")
medal_counts[medal_counts["event_medal"] > 1]

,discipline,event,event_medal,nb_rows
55,Curling,Men,3,1
58,Curling,Mixed Doubles,3,1
61,Curling,Women,3,1
76,Freestyle Skiing,Men's Dual Moguls,3,1
93,Freestyle Skiing,Women's Dual Moguls,3,1
106,Ice Hockey,Men,3,1
109,Ice Hockey,Women,3,1
167,Snowboard,Men's Parallel Giant Slalom,3,1
180,Snowboard,Women's Parallel Giant Slalom,3,1
198,Speed Skating,Men's Team Pursuit,3,1


In [135]:
event_medal_3 = df_schedules[df_schedules["event"] == "Women's Dual Moguls"]
event_medal_3

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
942,2026-02-14T10:30:00+01:00,2026-02-14T10:57:30+01:00,2026-02-14,FINISHED,Freestyle Skiing,FRS,Women's Dual Moguls,0,Women's Dual Moguls 1/16 Finals,W,INDV,Livigno Aerials & Moguls Park,LAM,Livigno Aerials & Moguls Park-Moguls & Dual Mo...,LMO,FRSWDM----------------R32---------
943,2026-02-14T11:00:00+01:00,2026-02-14T11:15:00+01:00,2026-02-14,FINISHED,Freestyle Skiing,FRS,Women's Dual Moguls,0,Women's Dual Moguls 1/8 Finals,W,INDV,Livigno Aerials & Moguls Park,LAM,Livigno Aerials & Moguls Park-Moguls & Dual Mo...,LMO,FRSWDM----------------8FNL--------
945,2026-02-14T11:20:00+01:00,2026-02-14T11:28:00+01:00,2026-02-14,FINISHED,Freestyle Skiing,FRS,Women's Dual Moguls,0,Women's Dual Moguls Quarterfinals,W,INDV,Livigno Aerials & Moguls Park,LAM,Livigno Aerials & Moguls Park-Moguls & Dual Mo...,LMO,FRSWDM----------------QFNL--------
946,2026-02-14T11:35:00+01:00,2026-02-14T11:39:00+01:00,2026-02-14,FINISHED,Freestyle Skiing,FRS,Women's Dual Moguls,0,Women's Dual Moguls Semifinals,W,INDV,Livigno Aerials & Moguls Park,LAM,Livigno Aerials & Moguls Park-Moguls & Dual Mo...,LMO,FRSWDM----------------SFNL--------
947,2026-02-14T11:46:00+01:00,2026-02-14T11:48:00+01:00,2026-02-14,FINISHED,Freestyle Skiing,FRS,Women's Dual Moguls,3,Women's Dual Moguls Finals,W,INDV,Livigno Aerials & Moguls Park,LAM,Livigno Aerials & Moguls Park-Moguls & Dual Mo...,LMO,FRSWDM----------------FNL-000200--
948,2026-02-14T11:48:00+01:00,2026-02-14T11:50:00+01:00,2026-02-14,FINISHED,Freestyle Skiing,FRS,Women's Dual Moguls,1,Women's Dual Moguls Finals,W,INDV,Livigno Aerials & Moguls Park,LAM,Livigno Aerials & Moguls Park-Moguls & Dual Mo...,LMO,FRSWDM----------------FNL-000100--


After checking all the schedules with 3 for "even_medal" have the same row except for "event_medal" which has 1. Hence these rows can be dropped!

Scheduled is the ceremony (see tail) which is also the reason for 17 "disciplines" even though only 16 took place. As one can see, cancelled events took place afterwards and are the reason for the NaN "end_date".

Conclusion:

Need to drop duplicates! (Most likely all rows were repeated three times)

All disciplines are present. One can assume complete sub-dataset (Some events may not have all their trainings hence using only medal events may be safer). Some events only have their medal event.

## Teams

In [137]:
duplicated_teams = df_teams[df_teams.duplicated()]
duplicated_teams

,code,name,team_type,gender,country_code,events


In [138]:
df_teams.nunique()

code            550
name            205
team_type         5
gender            3
country_code     43
events           39
dtype: int64

In [139]:
df_teams.shape

(550, 6)

In [140]:
df_teams.info()

<class 'pandas.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   code          550 non-null    str  
 1   name          550 non-null    str  
 2   team_type     550 non-null    str  
 3   gender        550 non-null    str  
 4   country_code  550 non-null    str  
 5   events        550 non-null    str  
dtypes: str(6)
memory usage: 25.9 KB


In [141]:
df_teams.head()

,code,name,team_type,gender,country_code,events
0,FSKXPAIRS---ARM01,AKOPOVA Karina / RAKHMANIN Nikita,CPLW,X,ARM,"[{'discipline': 'Figure Skating', 'event': 'Pa..."
1,BOBMTEAM2---GER03,AMMOUR Adam,CUSTOM,M,GER,"[{'discipline': 'Bobsleigh', 'event': '2-man'}]"
2,BOBMTEAM4---GER03,AMMOUR Adam,CUSTOM,M,GER,"[{'discipline': 'Bobsleigh', 'event': '4-man'}]"
3,BOBWTEAM2---ITA01,ANDREUTTI Giada,CUSTOM,W,ITA,"[{'discipline': 'Bobsleigh', 'event': '2-woman'}]"
4,SKNXRELAY2--LAT01,ANDZANE Marta / INDRIKSONS Emils,CPLW,X,LAT,"[{'discipline': 'Skeleton', 'event': 'Mixed Te..."


In [142]:
df_teams.tail()

,code,name,team_type,gender,country_code,events
545,BOBMTEAM2---NED01,WESSELINK Dave,CUSTOM,M,NED,"[{'discipline': 'Bobsleigh', 'event': '2-man'}]"
546,BOBWTEAM2---CHN02,YING Qing,CUSTOM,W,CHN,"[{'discipline': 'Bobsleigh', 'event': '2-woman'}]"
547,FSKXICEDANCEJPN01,YOSHIDA Utana / MORITA Masaya,CPLW,X,JPN,"[{'discipline': 'Figure Skating', 'event': 'Te..."
548,SKNXRELAY2--CHN02,ZHAO Dan / CHEN Wenhao,CPLW,X,CHN,"[{'discipline': 'Skeleton', 'event': 'Mixed Te..."
549,FSKXICEDANCEUSA02,ZINGAS Emilea / KOLESNIK Vadym,CPLW,X,USA,"[{'discipline': 'Figure Skating', 'event': 'Ic..."


In [143]:
df_teams.groupby("team_type").size()

team_type
CPLP       28
CPLW       59
CUSTOM     78
ORG       329
ORGN       56
dtype: int64

In [144]:
countries_athletes = set(df_athletes["country_code"])
countries_teams = set(df_teams["country_code"])

missing_countries = countries_athletes - countries_teams


missing_countries

{'AIN',
 'ALB',
 'AND',
 'AZE',
 'BEN',
 'BIH',
 'BOL',
 'CHI',
 'COL',
 'CYP',
 'ECU',
 'ERI',
 'GBS',
 'HAI',
 'HKG',
 'IND',
 'IRI',
 'IRL',
 'ISL',
 'KEN',
 'KGZ',
 'KOS',
 'KSA',
 'LBN',
 'LUX',
 'MAD',
 'MAR',
 'MAS',
 'MDA',
 'MEX',
 'MGL',
 'MKD',
 'MLT',
 'MNE',
 'MON',
 'NGR',
 'NZL',
 'PAK',
 'PHI',
 'POR',
 'PUR',
 'RSA',
 'SGP',
 'SMR',
 'SRB',
 'THA',
 'UAE',
 'URU',
 'UZB',
 'VEN'}

In [145]:
df_teams.groupby("gender").size()

gender
M    217
W    165
X    168
dtype: int64

Conclusion:

Incomplete sub-dataset. Missing countries as well as events.

Name is either individuals or country.

Limited potential use or use with great care. Requires further processing to complete gender data ("-" to "X"/"M"/"F") through "event" column which indicates gender for most cases.

## Venues

In [146]:
duplicated_venues = df_venues[df_venues.duplicated()]
duplicated_venues

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue,events


In [147]:
df_venues.nunique()

venue               15
venue_code          15
disciplines         12
disciplines_code    12
url_sport           12
url_venue           15
events              14
dtype: int64

In [148]:
df_venues.shape

(15, 7)

In [149]:
df_venues.info()

<class 'pandas.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   venue             15 non-null     str  
 1   venue_code        15 non-null     str  
 2   disciplines       13 non-null     str  
 3   disciplines_code  13 non-null     str  
 4   url_sport         13 non-null     str  
 5   url_venue         15 non-null     str  
 6   events            15 non-null     str  
dtypes: str(7)
memory usage: 972.0 bytes


In [150]:
df_venues.head()

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue,events
0,Anterselva Biathlon Arena,ABA,discipline-biathlon,BTH,['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Biathlon', 'Mixed Relay 4 x 6km (M+W)'), ('..."
1,Cortina Curling Olympic Stadium,CCU,discipline-curling,CUR,['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Curling', 'Mixed Doubles'), ('Curling', 'Me..."
2,Cortina Sliding Centre,CSC,"[""discipline-bobsleigh"",""discipline-luge"",""dis...","[""BOB"",""LUG"",""SKN""]",['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Luge', ""Men's Singles""), ('Luge', ""Women's ..."
3,Livigno Aerials & Moguls Park,LAM,discipline-freestyle-skiing,FRS,['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Freestyle Skiing', ""Men's Moguls""), ('Frees..."
4,Livigno Snow Park,LSP,"[""discipline-freestyle-skiing"",""discipline-sno...","[""FRS"",""SBD""]",['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Snowboard', ""Men's Snowboard Big Air""), ('F..."


In [151]:
df_venues.tail()

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue,events
10,Predazzo Ski Jumping Stadium,PSJ,"[""discipline-nordic-combined"",""discipline-ski-...","[""NCB"",""SJP""]",['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Ski Jumping', ""Women's Normal Hill""), ('Ski..."
11,Stelvio Ski Centre,SSC,"[""discipline-alpine-skiing"",""discipline-ski-mo...","[""ALP"",""SMT""]",['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Alpine Skiing', ""Men's Downhill""), ('Alpine..."
12,Tesero Cross-Country Skiing Stadium,TCC,"[""discipline-cross-country-skiing"",""discipline...","[""CCS"",""NCB""]",['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Cross-Country Skiing', ""Women's 10km + 10km..."
13,Tofane Alpine Skiing Centre,CAL,discipline-alpine-skiing,ALP,['https://www.olympics.com/en/milano-cortina-2...,https://www.olympics.com/en/milano-cortina-202...,"[('Alpine Skiing', ""Women's Downhill""), ('Alpi..."
14,Verona Olympic Arena,VOA,NaN,NaN,NaN,https://www.olympics.com/en/milano-cortina-202...,"[('Ceremonies', 'Closing Ceremony')]"


In [152]:
disciplines_set = set()

for disciplines in df_venues["disciplines"]:
    # Ignore NaN
    if pd.isna(disciplines):
        continue
    
    try:
        disciplines_list = ast.literal_eval(disciplines)   # convert string -> list (for multiple disciplines at venue otherwise crash)
        disciplines_set.update(disciplines_list)

    except (ValueError, SyntaxError):
        # If ast.literal_eval fails (case for single disciplines at venue), add the value directly
        disciplines_set.add(disciplines)

print("Total unique disciplines: " + str(len(disciplines_set)))
print(disciplines_set)

Total unique disciplines: 16
{'discipline-nordic-combined', 'discipline-speed-skating', 'discipline-curling', 'discipline-alpine-skiing', 'discipline-ski-jumping', 'discipline-luge', 'discipline-biathlon', 'discipline-cross-country-skiing', 'discipline-freestyle-skiing', 'discipline-ice-hockey', 'discipline-snowboard', 'discipline-figure-skating', 'discipline-short-track-speed-skating', 'discipline-ski-mountaineering', 'discipline-skeleton', 'discipline-bobsleigh'}


In [153]:
# Doing the same for disciplines_code
disciplines_set = set()

for disciplines in df_venues["disciplines_code"]:
    # Ignore NaN
    if pd.isna(disciplines):
        continue
    
    try:
        disciplines_list = ast.literal_eval(disciplines)   # convert string -> list (for multiple disciplines at venue otherwise crash)
        disciplines_set.update(disciplines_list)

    except (ValueError, SyntaxError):
        # If ast.literal_eval fails (case for single disciplines at venue), add the value directly
        disciplines_set.add(disciplines)

print("Total unique disciplines: " + str(len(disciplines_set)))
print(disciplines_set)

Total unique disciplines: 16
{'BOB', 'SJP', 'FSK', 'CUR', 'FRS', 'LUG', 'BTH', 'SKN', 'IHO', 'ALP', 'CCS', 'NCB', 'SMT', 'STK', 'SBD', 'SSK'}


In [154]:
df_venues[df_venues["disciplines"].isna()]

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue,events
6,Milano San Siro Olympic Stadium,MOS,NaN,NaN,NaN,https://www.olympics.com/en/milano-cortina-202...,[]
14,Verona Olympic Arena,VOA,NaN,NaN,NaN,https://www.olympics.com/en/milano-cortina-202...,"[('Ceremonies', 'Closing Ceremony')]"


Milano San Siro Olympic Stadium was used for Opening ceremony of the Olympics.

Verona Olympic Arena was used for Closing ceremony of the Olympics (and Paralympic opening ceremony).


Conclusion:

Near complete sub-dataset. Easy to complete.

Would need preprocessing to determine exact events (sub-disciplines which can take place in two different places).

Would also need to complete sport url in pre-processing

# Ideas of exploration

Local map with venues followed by events clickable to enter new page dedicated to sports which took place there

Look into past winter olympics and see dynasties/new comers in Milan 2026 (require extra dataset)

Look into visitors/economical/environmentals impacts (require extra dataset)

Do animation where we see dots arriving to the games (1 dot per athlete): start animation

Classic medal race with numerous filters (discipline/event/country region/gender/team event vs. individual vs. duo etc.)

Flag bearers dedicated page (legendary figures)

Men vs Women in discipline per country

Team sports per country

Most exploited venues (use schedules.csv/venues.csv)

Which sports have most medals (where to specialize if medal race is the goal)

Use sports/venue URL to redirect to Olympics official website

Don't forget to cite dataset!

*["Milano-Cortina 2026 Olympic Winter Games"](https://www.kaggle.com/datasets/piterfm/milano-cortina-2026-olympic-winter-games/data?select=medals.csv)*